src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,581 @@
+        controlOperationsLock.lock();
+        try {
+            refreshSshConnection();
+            response = send(command);
+            int retryAttempts = 0;
+            while (!response.contains(verifyString)) {
+                retryAttempts++;
+                response = send(command);
+                if (retryAttempts >= 10) {
+                    throw new IllegalStateException(String.format("Failed to verify response for command %s. Expected output: %s", command, verifyString));
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return response;
+    }
+
+    /**
+     * Check if the channel is connected and if not - reconnect
+     */
+    private void refreshSshConnection() throws Exception {
+        if (!isChannelConnected()) {
+            connect();
+        }
+    }
+
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        if (StringUtils.isNullOrEmpty(dialDevice.getDialString())) {
+            throw new InvalidArgumentException("Dial string is empty.");
+        }
+        String meetingStatus = getCallStatus();
+        if (meetingStatus.equals(IN_MEETING)) {
+            if(logger.isWarnEnabled()) {
+                logger.warn("Not able to connect. Meeting " + meetingNumber + " is in progress.");
+            }
+            return null;
+        } else if (meetingStatus.equals(CONNECTING_MEETING)) {
+            // Need to be able to recover if terminal stucks in a "connecting" state.
+            // This may happen if some specific non-existing meeting numbers are used (this may happen by accident)
+            // in which case ZR app will stuck without giving an ability to disconnect (unless it's done manually)
+            // So it is assumed that if we're getting here - it's either multiple calls are being addressed one right
+            // after another, which is less likely, or that the terminal got stuck because of this issue.
+            callDisconnect();
+        }
+        return joinStartMeeting(dialDevice.getDialString());
+    }
+
+    @Override
+    public void hangup(String s) throws Exception {
+        String meetingStatus = getCallStatus();
+        if (!meetingStatus.equals(IN_MEETING) && !meetingStatus.equals(CONNECTING_MEETING)) {
+            return;
+        }
+        callDisconnect();
+    }
+
+    @Override
+    public CallStatus retrieveCallStatus(String s) throws Exception {
+        CallStatus callStatus = new CallStatus();
+        callStatus.setCallStatusState(getCallStatus().equals(IN_MEETING) ? CallStatus.CallStatusState.Connected : CallStatus.CallStatusState.Disconnected);
+        callStatus.setCallId(meetingNumber);
+        return callStatus;
+    }
+
+    @Override
+    public MuteStatus retrieveMuteStatus() throws Exception {
+        String meetingStatus = getCallStatus();
+        if (!meetingStatus.equals(IN_MEETING)) {
+            return MuteStatus.Unmuted;

```

Comment:
This is wrong logically. The fact that room is not in the meeting does not logically mean that it is unmuted.
Whenever code does not follow logic, it has to be documented why.

Reply:
Adding details, it's not possible retrieve/change status while not in the call, because both operations use the same CLI command